### PR TITLE
Remove Adapt dependence, use ETOPO2022 dataset and regrid orography using SpaceVaryingInputs

### PR DESCRIPTION
**Pull request by akshaysridhar**

### CODE REVIEW

**Comment by akshaysridhar** [comment]
32x_Biharmonic_Orography
256x_Biharmonic_Orography
Biharmonic_smoothing_0-256pass
Laplacian_0-256pass

**Comment by Sbozzolo** [comment]
Please, also remove the `gaussian_smooth`, its tests, and the dependency on Distributions in `test`, if it that's no longer needed.

File: examples/topography_spectra.jl
Comment (Sbozzolo) [inline examples/topography_spectra.jl:1]: Can you add some documentation to this example?

File: src/utils/AtmosArtifacts.jl [hunk 1]
```diff
@@ -67,4 +67,15 @@ function aerosol_concentration_file_path(; context = nothing)
...
 
+"""
+    earth_orography_file_path(; context=nothing)
+
+Construct the file path for the 60arcsecond orography data NetCDF file.
+
+Downloads the 60arc-second dataset by default. 
+"""
+function earth_orography_file_path(; context = nothing)
+    return res_file_path("earth_orography_60arcseconds"; context)
```
Comment (Sbozzolo) [inline src/utils/AtmosArtifacts.jl:78]:
```suggestion
    full_name = "earth_orography_60arcseconds"
    return joinpath(@clima_artifact(full_name, context), "$(full_name).nc")
```

File: src/utils/common_spaces.jl [hunk 2]
```diff
@@ -74,6 +74,38 @@ function make_horizontal_space(mesh, quad, comms_ctx, bubble)
     return space
 end
 
+function diffuse_surface_elevation_biharmonic!(
```
Comment (Sbozzolo) [inline src/utils/common_spaces.jl:77]: Can you add docstring to this function, including references to where the equations came from?

File: src/utils/common_spaces.jl [hunk 3]
```diff
@@ -100,8 +132,39 @@ function make_hybrid_spaces(
...
+        @info "SpaceVaryingInputs: Remapping orography onto spectral space"
+        z_surface = SpaceVaryingInputs.SpaceVaryingInput(
+            AA.earth_orography_file_path(;
+                context = ClimaComms.context(h_space),
+            ),
+            "z",
+            h_space,
+        )
+        parent(z_surface) .=
+            ifelse.(parent(z_surface) .< FT(0), FT(0), parent(z_surface))
```
Comment (Sbozzolo) [inline src/utils/common_spaces.jl:147]: Maybe we should apply this to the input data directly? You can pass a similiar preprocess func to the `SpaceVaryingInput`

File: src/utils/common_spaces.jl [hunk 4]
```diff
@@ -100,8 +132,39 @@ function make_hybrid_spaces(
...
+            hypsography = Hypsography.SLEVEAdaption(
+                Geometry.ZPoint.(z_surface),
+                FT(parsed_args["sleve_eta"]),
+                FT(parsed_args["sleve_s"]),
+            )
+        elseif parsed_args["mesh_warp_type"] == "Linear"
+            @info "Linear mesh warp"
+            hypsography =
+                Hypsography.LinearAdaption(Geometry.ZPoint.(z_surface))
+        end
```
Comment (Sbozzolo) [inline src/utils/common_spaces.jl:167]: What happens if `parsed_args["mesh_warp_type"] == "linear"`? Or some other cases that are not captured by this if-else?

File: src/solver/type_getters.jl [hunk 5]
```diff
@@ -183,18 +164,27 @@ function get_spaces(parsed_args, params, comms_ctx)
...
             make_hybrid_spaces(
                 h_space,
                 z_max,
                 z_elem,
                 z_stretch;
-                parsed_args = parsed_args,
                 surface_warp = warp_function,
                 deep,
             )
+        elseif topography == "Earth"
```
Comment (szy21) [inline src/solver/type_getters.jl:178]: We can first check topography here, then check warp_function, and can get rid of the `&& topography != "Earth"`? I think the logic is more clear this way.

File: src/utils/common_spaces.jl [hunk 6]
```diff
@@ -100,8 +132,39 @@ function make_hybrid_spaces(
         z_mesh,
     )
     z_grid = Grids.FiniteDifferenceGrid(z_topology)
-    if isnothing(surface_warp)
+    if isnothing(surface_warp) && parsed_args["topography"] != "Earth"
         hypsography = Hypsography.Flat()
+    elseif isnothing(surface_warp) && parsed_args["topography"] == "Earth"
```
Comment (szy21) [inline src/utils/common_spaces.jl:137]: Same here, we can check topography first then surface_warp to make the logic more clear.

File: examples/topography_spectra.jl
Comment (szy21) [reply examples/topography_spectra.jl:1]: If I understand correctly, this example is to create a space with different iterations of Laplacian smoothing and plot the spectra. Is there a way to plot the spectra from topography diagnostics from a simulation? I think the more general use case would be someone runs a global simulation and wants to see the spectra of that particular simulation, without generating the space again.

**Review by szy21 — COMMENTED** [review]
Thanks! Here are some comments. And I agree that some documentation would be helpful.

File: examples/topography_spectra.jl
Comment (akshaysridhar) [reply examples/topography_spectra.jl:1]: Yes; we've had such a plot in our CI pipelines before (for the DCMIP case) - `orog.nc` allows us to generate this plot as part of normal CI builds. I'll reintroduce this.

File: src/solver/type_getters.jl (hunk 5 above)
Comment (akshaysridhar) [reply src/solver/type_getters.jl:178]: Yes ; I'll go over this again and try to clean up some of these branching statements in `common_spaces` and `type_getters.`

File: src/utils/common_spaces.jl (hunk 2 above)
Comment (akshaysridhar) [reply src/utils/common_spaces.jl:77]: Sure; I can add better docstrings; these functions are implementations of unsteady diffusion equations applied to the surface orography.

Comment (akshaysridhar) [reply src/utils/common_spaces.jl:77]: @Sbozzolo I'll move this function directly to ClimaCore (within the Hypsography module)

File: examples/Manifest.toml [hunk 7]
```diff
@@ -1,8 +1,8 @@
 # This file is machine-generated - editing it directly is not advised
 
-julia_version = "1.10.5"
+julia_version = "1.10.4"
```
Comment (szy21) [inline examples/Manifest.toml:3]: Could we revert this change?

File: src/solver/type_getters.jl [hunk 8]
```diff
@@ -183,17 +164,27 @@ function get_spaces(parsed_args, params, comms_ctx)
         else
             Meshes.Uniform()
         end
-        if warp_function == nothing
-            make_hybrid_spaces(h_space, z_max, z_elem, z_stretch; deep)
-        else
+        if topography == "Earth"
```
Comment (szy21) [inline src/solver/type_getters.jl:167]:
```suggestion
        if warp_function == nothing
```

Comment (szy21) [reply src/solver/type_getters.jl:167]: (It seems this and the following branches are returning the same thing. They both return `make_hybrid_spaces` with `surface_warp=nothing`.)

File: src/solver/type_getters.jl [hunk 9]
```diff
@@ -183,17 +164,27 @@ function get_spaces(parsed_args, params, comms_ctx)
...
                 h_space,
                 z_max,
                 z_elem,
                 z_stretch;
+                surface_warp = nothing,
                 parsed_args = parsed_args,
+                deep,
+            )
+        elseif warp_function == nothing && topography != "Earth"
+            make_hybrid_spaces(h_space, z_max, z_elem, z_stretch; deep, parsed_args)
```
Comment (szy21) [inline src/solver/type_getters.jl:178]:
```suggestion
```

File: src/utils/common_spaces.jl [hunk 10]
```diff
@@ -100,7 +100,42 @@ function make_hybrid_spaces(
...
+            ),
+            "z",
+            h_space,
+        )
+        parent(z_surface) .=
+            ifelse.(parent(z_surface) .< FT(0), FT(0), parent(z_surface))
+        Δh_scale = Spaces.node_horizontal_length_scale(h_space)
+        Hypsography.diffuse_surface_elevation!(
+            z_surface;
+            κ = FT((Δh_scale)^2 / 100),
```
Comment (szy21) [inline src/utils/common_spaces.jl:118]: Could you add a comment on where this coefficient comes from?

File: src/solver/type_getters.jl [hunk 11]
```diff
@@ -146,26 +146,7 @@ function get_spaces(parsed_args, params, comms_ctx)
...
-                linear_interpolation(
-                    (lon, lat),
-                    esmth,
-                    extrapolation_bc = (Periodic(), Flat()),
-                ),
-            )
-        end
-        @info "Generated interpolation stencil"
-        warp_function = generate_topography_warp(earth_spline)
+        warp_function = nothing
```
Comment (szy21) [inline src/solver/type_getters.jl:149]: Is there a reason for setting warp_function to nothing for Earth topography, and using parsed_args["topography"] to set z_surface in `make_hybrid_spaces`? I find it a bit confusing that warp_function is nothing for Earth topography.

Comment (akshaysridhar) [reply src/solver/type_getters.jl]: The previous iteration was an attempt to keep this a non-breaking change; I've updated the `make_hybrid_space` function to use `SpaceVaryingInputs` everywhere which simplifies some of these branches.

**Comment by Sbozzolo** [comment]
@akshaysridhar could also remove the ETOPO1 artifact as part of this pull request? 

If it is still needed for tests, could you ensure that it is only downloaded in tests?

I think this is the only artifact left that ClimaCoupler has manually download, everything else is handled automatically.

**Comment by juliasloan25** [comment]
Is this PR ready to merge?

**Comment by akshaysridhar** [comment]
@szy21 There is a gravity wave function that uses the old artifact e.g. `gfdl_restart` vs `raw_topo` in `orography_gravity_wave` (without smoothing) - is it reasonable to update this here as well ? (I can separate that if needed)

**Comment by szy21** [comment]
> @szy21 There is a gravity wave function that uses the old artifact e.g. `gfdl_restart` vs `raw_topo` in `orography_gravity_wave` (without smoothing) - is it reasonable to update this here as well ? (I can separate that if needed)

Feel free to update it if the test doesn't break and the result looks fine.

File: src/utils/common_spaces.jl (hunk 10 above)
Comment (akshaysridhar) [reply src/utils/common_spaces.jl]: Updated, E3SM documentation guides this (otherwise it can be estimated from the grid resolutions and some nominal CFL)

**Comment by akshaysridhar** [comment]
> @akshaysridhar could also remove the ETOPO1 artifact as part of this pull request?
> 
> If it is still needed for tests, could you ensure that it is only downloaded in tests?
> 
> I think this is the only artifact left that ClimaCoupler has manually download, everything else is handled automatically.

@Sbozzolo if the changes look reasonable here we can prepare to remove the Coupler download (as in the land-sea mask PR).

File: examples/topography_spectra.jl [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+to apply `diffiter` instances of Laplacian diffusion passes 
+(which rely on the canonical unsteady diffusion equation).
+Returns a `Field` containing surface orography. 
+"""
+function generate_spaces(; h_elem = 60, diffiter = 32)
+    FT = Float32
+    param_dict = CP.create_toml_dict(FT)
+    params = CP.get_parameter_values(param_dict, ["planet_radius"])
+    cubed_sphere_mesh =
+        CA.cubed_sphere_mesh(; radius = params.planet_radius, h_elem)
```
Comment (Sbozzolo) [inline examples/topography_spectra.jl:32]:
```suggestion
function generate_spaces(; h_elem = 60, diffiter = 32, planet_radius = 6.378e6)
    FT = Float32
    cubed_sphere_mesh =
        CA.cubed_sphere_mesh(; radius = FT(planet_radius), h_elem)
```
I think it is an overkill to have this function depending on `ClimaParams` when all we need is an Earth radius. This is also just an example.

File: examples/topography_spectra.jl [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+    quad = Quadratures.GLL{4}()
+    comms_ctx = ClimaComms.SingletonCommsContext()
```
Comment (Sbozzolo) [inline examples/topography_spectra.jl:34]: Why is this restriction?

File: examples/topography_spectra.jl [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+    @assert h_space isa CC.Spaces.SpectralElementSpace2D
+    coords = CC.Fields.coordinate_field(h_space)
+    target_field = CC.Fields.zeros(h_space)
+    elev_from_file = SpaceVaryingInputs.SpaceVaryingInput(
+        AA.earth_orography_file_path(; context = comms_ctx),
+        "z",
+        h_space,
+    )
+    parent(elev_from_file) .=
+        ifelse.(parent(elev_from_file) .< FT(0), FT(0), parent(elev_from_file))
```
Comment (Sbozzolo) [inline examples/topography_spectra.jl:46]: You can avoid using `parent` by using a mask like this:
```julia
function mask(x::FT) where {FT}
     return FT(x > 0)
end
```

File: examples/topography_spectra.jl [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+    diffuse_surface_elevation!(
+        elev_from_file,
+        κ = FT((Δh_scale)^2 / 100),
+        dt = FT(1),
+        maxiter = diffiter,
+    )
+    parent(elev_from_file) .=
+        ifelse.(parent(elev_from_file) .< FT(0), FT(0), parent(elev_from_file))
```
Comment (Sbozzolo) [inline examples/topography_spectra.jl:54]: Same as before

File: examples/topography_spectra.jl [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+end
+
+"""
+    remap_to_array(var, hcoords)
+Given an Array of `hcoords`, this function remaps a ClimaCore
+Field `var` onto `hcoords`. 
+Returns an `Array` containing the values which make up `var`.
+"""
+function remap_to_array(var, hcoords)
+    remapper = Remapping.Remapper(CC.axes(var), hcoords)
```
Comment (Sbozzolo) [inline examples/topography_spectra.jl:65]:
```suggestion
    remapper = Remapping.Remapper(axes(var), hcoords)
```

File: examples/topography_spectra.jl [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+
+"""
+    generate_all_spectra(;h_elem=60)
+Uses the spectral calculator and space generation tools in this example file
+to generate a series of surface orography fields with different extents of 
+Laplacian smoothing. 
+Returns a `Plots.Figure`. 
+"""
+function generate_all_spectra(; h_elem = 60)
+    fig = Plots.plot()
```
Comment (Sbozzolo) [inline examples/topography_spectra.jl:108]: Would it be possible to use Makie instead?

File: src/parameterized_tendencies/gravity_wave_drag/orographic_gravity_wave.jl [hunk 18]
```diff
@@ -39,7 +39,7 @@ function orographic_gravity_wave_cache(Y, ogw::OrographicGravityWave)
     elseif ogw.topo_info == "raw_topo"
         # TODO: right now this option may easily crash
         # because we did not incorporate any smoothing when interpolate back to model grid
-        elevation_rll = joinpath(topo_elev_dataset_path(), "ETOPO1_coarse.nc")
+        elevation_rll = AA.earth_orography_file_path(; context=ClimaComms.context(Spaces.horizontal_space(axes(Y.c))))
```
Comment (Sbozzolo) [inline src/parameterized_tendencies/gravity_wave_drag/orographic_gravity_wave.jl:42]:
```suggestion
        elevation_rll = AA.earth_orography_file_path(; context=ClimaComms.context(Y.c)))
```
I think that this might be enough

File: src/solver/type_getters.jl [hunk 19]
```diff
@@ -183,19 +149,14 @@ function get_spaces(parsed_args, params, comms_ctx)
...
-                deep,
-            )
-        end
+        make_hybrid_spaces(
+            h_space,
+            z_max,
+            z_elem,
+            z_stretch;
+            deep,
+            parsed_args = parsed_args,
```
Comment (Sbozzolo) [inline src/solver/type_getters.jl:158]:
```suggestion
            parsed_args,
```

File: src/utils/common_spaces.jl (hunk 3 above)
Comment (Sbozzolo) [reply src/utils/common_spaces.jl:132]: Same as above with regards to avoiding using `parent`

File: src/utils/common_spaces.jl [hunk 20]
```diff
@@ -100,11 +98,66 @@ function make_hybrid_spaces(
...
+        # E3SM
+        # https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/1456603764/V1+Topography+GLL+grids
+        Hypsography.diffuse_surface_elevation!(
+            z_surface;
+            κ = FT(28e7),
+            dt = FT(1),
+            maxiter = 128,
+        )
+        parent(z_surface) .=
+            ifelse.(parent(z_surface) .< FT(0), FT(0), parent(z_surface))
```
Comment (Sbozzolo) [inline src/utils/common_spaces.jl:145]: Here too

File: examples/topography_spectra.jl (hunk 12 above)
Comment (akshaysridhar) [reply examples/topography_spectra.jl:32]: Sure, this was intended as a quick demo of `params` usage in `CA` (I realise that the `ClimaParams` documentation also has this. I'll change this to just depend on the keyword arg.

File: examples/topography_spectra.jl (hunk 13 above)
Comment (akshaysridhar) [reply examples/topography_spectra.jl:32]: This example is light enough that it can be run on a single device, so MPI / generalized context was unnecessary. I'll generalize with `ClimaComms.context()`.

**Comment by akshaysridhar** [comment]
@Sbozzolo Masks to avoid `parent`, `Makie`, and `context` have been fixed following your suggestions.

File: examples/topography_spectra.jl [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+
+const AA = AtmosArtifacts
+
+
+function mask(x::FT) where {FT}
+    return x * FT(x > 0)
+end
+
+"""
+    generate_spaces(; h_elem=60, diffiter=32)
```
Comment (Sbozzolo) [inline examples/topography_spectra.jl:23]:
```suggestion
    generate_spaces(; h_elem=60, diffiter=32, planet_radius = 6.378e6)
```

**Review by Sbozzolo — APPROVED** [review]

**Comment by Sbozzolo** [comment]
Thank you!

**Comment by akshaysridhar** [comment]
@Sbozzolo @szy21 `compute hmax` in the gravity wave parameterization is expensive with the new dataset (ci timeouts) so I added a simple downscaled array (Skip `nth` pt in data so array size is `3600x1800`).

File: examples/topography_spectra.jl [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+For a given number of elements `h_elem`, this function generates a 
+horizontal spectral space (ClimaCore.SpectralElementSpace2D) and
+generates an interpolated (using SpaceVaryingInputs) orography field
+on this space. The `diffuse_surface_elevation` function is called 
+to apply `diffiter` instances of Laplacian diffusion passes 
+(which rely on the canonical unsteady diffusion equation).
```
Comment (tapios) [inline examples/topography_spectra.jl:29]:
```suggestion
(which rely on the canonical diffusion equation).
```

File: examples/topography_spectra.jl [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+    target_field = CC.Fields.zeros(h_space)
+    elev_from_file = SpaceVaryingInputs.SpaceVaryingInput(
+        AA.earth_orography_file_path(; context = comms_ctx),
+        "z",
+        h_space,
+    )
+    elev_from_file = @. mask(elev_from_file)
+    diffuse_surface_elevation!(
+        elev_from_file,
+        κ = FT(28e7 * (30 / h_elem)^2),
```
Comment (tapios) [inline examples/topography_spectra.jl:51]:
```suggestion
        κ = diff_courant * Δh_scale^2 / dt,
```
Please avoid arbitrary numbers like 28e7 in code. I think what you want here is something like the above, where diff_courant is some parameter to ensure numerical stability (e.g., 0.1).

File: examples/topography_spectra.jl [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+        dt = FT(1),
+        maxiter = diffiter,
```
Comment (tapios) [inline examples/topography_spectra.jl:53]: As I mentioned on Slack, I think it's better to compute `maxiter` as `log(n) / diff_courant` where `n` is the fractional damping of the smallest resolved scale and would be a good input parameter (e.g., `n = 1000` or so).

File: examples/topography_spectra.jl [hunk 25]
```diff
@@ -0,0 +1,129 @@
...
+"""
+function remap_to_array(test_var, hcoords)
+    remapper = Remapping.Remapper(axes(test_var), hcoords)
+    orog = Array(Remapping.interpolate(remapper, test_var))
+end
+
+"""
+    gen_spectra(test_var)
+Given a Field `test_var`, this function first calls `remap_to_array`
+to remap this spectral element field onto Gaussian long-lat
```
Comment (tapios) [inline examples/topography_spectra.jl:73]:
```suggestion
to remap this spectral element field onto a latitude-longitude
```
(it's not Gaussian)

File: examples/topography_spectra.jl [hunk 26]
```diff
@@ -0,0 +1,129 @@
...
+coordinates, and computes the spectra.
```
Comment (tapios) [inline examples/topography_spectra.jl:74]:
```suggestion
grid, and computes the spectra.
```

File: examples/topography_spectra.jl [hunk 27]
```diff
@@ -0,0 +1,129 @@
...
+Returns (`w_numbers`, `power_spectrum`) which contain the spherical
+wavenumbers and the spectral energy corresponding to this array
+of `w_numbers`. 
+"""
+function gen_spectra(test_var)
+    # Remap onto 1° resolution horizontal coordinates. 
+    longpts = range(-180.0, 180.0, 360)
+    latpts = range(-90.0, 90.0, 180)
+    hcoords =
```
Comment (tapios) [inline examples/topography_spectra.jl:83]: The fixed resolution of this grid will create problems, e.g., for high-resolution topography. Please replace by a grid that adapts to the model resolution, e.g., with a grid spacing of 360*Δh_scale/(2*pi earth_radius).

File: src/utils/common_spaces.jl [hunk 28]
```diff
@@ -100,11 +98,65 @@ function make_hybrid_spaces(
...
+        mask(x::FT) where {FT} = x * FT(x > 0)
+        z_surface = @. mask(z_surface)
+        # Coefficient for horizontal diffusion of surface orography
+        # is obtained from the V1, V2 Topography documentation from 
+        # E3SM
+        # https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/1456603764/V1+Topography+GLL+grids
+        nelem = Meshes.n_elements_per_panel_direction(axes(z_surface))
+        Hypsography.diffuse_surface_elevation!(
+            z_surface;
+            κ = FT(28e7 * (30 / nelem)^2),
```
Comment (tapios) [inline src/utils/common_spaces.jl:140]: Replace analogous to the above formulation.

File: examples/topography_spectra.jl (hunk 23 above)
Comment (akshaysridhar) [reply examples/topography_spectra.jl:51]: Noted, this value is from the E3SM docs which I can make clearer - in general I'll replace this (via changes in `ClimaCore` to depend entirely on resolution and required attenuation factors) - (following the Slack discussion)

**Comment by szy21** [comment]
Which one is the parameter / config that allows the user to choose damping factor? Or are you going to add it in the next PR?

**Comment by akshaysridhar** [comment]
> Which one is the parameter / config that allows the user to choose damping factor? Or are you going to add it in the next PR?

In this instance, there is a parameter `n_attenuation` ; the entire smoothing method should, in my opinion, be in `ClimaCore` which requires an additional package release + update so I want to get the artifact changes with a simple implementation into `ClimaAtmos` through this PR first.

**Comment by szy21** [comment]
> > Which one is the parameter / config that allows the user to choose damping factor? Or are you going to add it in the next PR?
> 
> In this instance, there is a parameter `n_attenuation` ; the entire smoothing method should, in my opinion, be in `ClimaCore` which requires an additional package release + update so I want to get the artifact changes with a simple implementation into `ClimaAtmos` through this PR first.

But `n_attenuation` is hard coded and not in config yet? I'm fine with getting this in, just to make sure.

**Comment by akshaysridhar** [comment]
> > > Which one is the parameter / config that allows the user to choose damping factor? Or are you going to add it in the next PR?
> > 
> > 
> > In this instance, there is a parameter `n_attenuation` ; the entire smoothing method should, in my opinion, be in `ClimaCore` which requires an additional package release + update so I want to get the artifact changes with a simple implementation into `ClimaAtmos` through this PR first.
> 
> But `n_attenuation` is hard coded and not in config yet? I'm fine with getting this in, just to make sure.

Right, it is currently defined only within the `common_spaces` util

**Comment by akshaysridhar** [comment]
> > > > Which one is the parameter / config that allows the user to choose damping factor? Or are you going to add it in the next PR?
> > > 
> > > 
> > > In this instance, there is a parameter `n_attenuation` ; the entire smoothing method should, in my opinion, be in `ClimaCore` which requires an additional package release + update so I want to get the artifact changes with a simple implementation into `ClimaAtmos` through this PR first.
> > 
> > 
> > But `n_attenuation` is hard coded and not in config yet? I'm fine with getting this in, just to make sure.
> 
> Right, it is currently defined only within the `common_spaces` util

@szy21: With the restart check updated in this commit, I've also added the damping factor to the config arg list.

File: test/restart.jl [hunk 29]
```diff
@@ -69,6 +69,11 @@ function _error(
     arr2::AbstractArray;
     ABS_TOL = 100eps(eltype(arr1)),
 )
+    # There are some parameters, e.g. Obukhov length, for which Inf
+    # is a reasonable value (implying a stability parameter in the neutral boundary layer
+    # regime, for instance). We account for such instances with the `isfinite` function.
+    arr1 .*= isfinite.(arr1)
```
Comment (akshaysridhar) [inline test/restart.jl:75]: cc @Sbozzolo

Comment (Sbozzolo) [reply test/restart.jl:75]: Looks like a good solution to me!
